### PR TITLE
Handle errors correctly when data lookup fails during /Pages-tree parsing (issue 14303)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This only applies to severely corrupt documents, where it's possible that the `Parser` throws when we try to access e.g. a /Kids-entry in the /Pages-tree.

Fixes two of the issues listed in issue #14303, namely the `poppler-742-0.pdf...` and `poppler-937-0.pdf...` documents.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
*The commit message was updated slightly, however no actual code changes were made.*

File: src/core/catalog.js [hunk 1]
```diff
@@ -1161,7 +1167,14 @@ class Catalog {
           }
         }
 
-        const kids = currentNode.get("Kids");
+        let kids;
+        try {
+          kids = currentNode.get("Kids");
```
Comment (calixteman) [inline src/core/catalog.js:1172]: Can't the other `currentNode.get("Type")` at line 1183 fail too ?

Comment (Snuffleupagus) [reply src/core/catalog.js:1172]: In theory it obviously could, even if we've *thus far* not seen such a document; anyway I've updated the patch as suggested.

**Review by calixteman — APPROVED** [review]
LGTM

**Pull request merged**